avocado/core/test.py

```diff
@@ -383,6 +384,11 @@ def _start_logging(self):
                                                        formatter,
                                                        self._ssh_logfile)
 
+        if isinstance(sys.stdout, output.LoggingFile):
+            sys.stdout.add_logger(logging.getLogger("avocado.test.stdout"))
+        if isinstance(sys.stderr, output.LoggingFile):
+            sys.stderr.add_logger(logging.getLogger("avocado.test.stderr"))
```

Comment:
^^ not literally, but the idea is to define a proper handler where the logger is defined.

Anyway the cleanest solution is still to add the `stdout`/`stderr` in `_start_logging` and remove them in `_stop_logging`. But it is important to remove them afterwards, otherwise the bug is still there.

Also I don't think we should include the first commit as it's not necessary, because the `stderr` when not forwarded won't cause the redirection.